packages/style-types/src/shared.ts

```diff
@@ -1,3 +1,4 @@
 export type GriffelStylesCSSValue = string | 0;
 
 export type ValueOrArray<T> = T | Array<T>;
+export type Fallback<T> = { [P in keyof T]: T[P] | NonNullable<T[P]>[] };
```

Comment:
can we document what's the purpose of this ?

```suggestion
/**
 * instead of using `CSS.PropertiesFallback` from csstype that remaps properties to `readonly` union we need to use our cutom Fallback mapped type in order to avoid type issues
 */
export type Fallback<T> = { [P in keyof T]: T[P] | NonNullable<T[P]>[] };
```